main.py

```diff
@@ -1,13 +1,144 @@
+# std libraries
+import time
+
+# external libraries
 from blessed import Terminal
 
-from library import Board
+# custom libraries
+from library.board import Board
+from library.user_term import UserTermState, starting_user_section, update_user_section
 
 term = Terminal()
 board = Board()
 
 with term.fullscreen(), term.cbreak(), term.hidden_cursor():
     board.draw_board(term)
 
+    # origin of user term
+    print(term.move_xy(0, 18))
+
+    # State
+    state = UserTermState()
+    player_active = 1
+    state.wait_for_ready = True
+
+    val = ""
+    term_info = [""] * 3
+
+    user_section = starting_user_section()
+    print("".join(user_section))
+
+    state.save_subgrid = True
+
     while (val := term.inkey()) != "q":
-        if val in ("1", "2", "3", "4", "5", "6", "7", "8", "9"):
-            continue
+        with term.location():
```

Comment:
What if we replace this `term.location()` with a `print(term.move_xy(*ORIGIN_OF_USER_TERM))`? That way, the things we call can use a `term.location()` context manager!

Note: I'm unsure this approach will work, but it's probably worth a shot.